Add basic Java generator usage instructions

andreaTP opened the pull request:
## Description

Basic usage instructions for the `java-generator`

## Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

sunix commented on doc/java-generator.md:
```diff
@@ -0,0 +1,85 @@
+# Java generation (preview)
```
Maybe rename it `Java generation from CRD`.
Also I would be consistent with the file name `java-generation-from-crd.md`

andreaTP replied:
ok

sunix commented on doc/java-generator.md:
```diff
@@ -0,0 +1,85 @@
+    </execution>
+  </executions>
+  <configuration>
+    ...
```
it would be nice to have a concrete example of  maven xml configuration with `source` and `target`

sunix replied:
or is the `source` be only defined with java properties?

andreaTP replied:
added `source` as `target` is always optional and already refers to a meaningful default.

sunix commented:
@andreaTP maybe not related to this PR, but I got some `package javax.validation.constraints does not exist` compilation errors. I am wondering if these constraints annotation were required and if it was possible to skip them in the generation?

andreaTP commented:
@sunix validation is currently mandatory, actually, I will add a section on the needed dependencies to compile the generated sources 👍 , along those lines:

https://github.com/fabric8io/kubernetes-client/blob/855fae6a15c0e471c1aba3b8359525a84c0ab993/java-generator/it/pom.xml#L46-L60

andreaTP commented:
here you go: https://github.com/fabric8io/kubernetes-client/pull/4219/commits/bcc8986d7775cc366d0d5a27903c145d7a03aece

sunix commented:
OK Thanks @andreaTP
I have another questions: it is generating a class that `extends io.fabric8.kubernetes.client.CustomResource<java.lang.Void, java.lang.Void>` is there a way to tell him to use my Spec and Status class?

sunix reviewed (approved):
It looks good to me. Then I am wondering if it makes sense to have this in the main `README.md` file, or at least a reference to it.

manusa commented on doc/java-generation-from-CRD.md:
```diff
@@ -0,0 +1,123 @@
+<plugin>
+  <groupId>io.fabric8</groupId>
+  <artifactId>java-generator-maven-plugin</artifactId>
+  <version><version></version>
```
```suggestion
  <version>${kubernetes-client.version}</version>
```
Maybe this line is confusing.

I'd replace this `<version>` value since the resulting XML is invalid and it's unclear.

manusa reviewed (changes requested):
Thx! LGTM besides the nit comments

andreaTP commented:
- automated the retrieval of the client version
- added the links in the main `Readme`
- addressed hopefully all the requests

squashed the commits and ready for final review 👍

sunix reviewed (approved):
LGTM

manusa reviewed (approved):
LGTM, thx!
